pkgs/by-name/tr/tritty/package.nix

```diff
@@ -0,0 +1,43 @@
+{ callPackage, fetchFromGitHub, lib, stdenv }:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "tritty";
+  # The number after + indicates the number of commits since the release.
+  version = "1.0+12";
+  src = fetchFromGitHub {
+    owner = "sjmulder";
+    repo = "trickle";
+    rev = "57bbf836a232d931404b66938941b2571fdc716f";
+    hash = "sha256-D9UCryE3+rODsP4rXVJrE2gL44EVSQ6fN3doQXJkbJE=";
+  };
+  makeFlags = [ "PREFIX=${builtins.placeholder "out"}" ];
+
+  passthru = {
+    tests = {
+      simple = callPackage ./tests/simple.nix { tritty = finalAttrs.finalPackage; };
+    };
+  };
+
+  meta = {
+    description = "Slow pipe and terminal";
+    license = lib.licenses.bsd2;
+    homepage = "https://github.com/sjmulder/trickle#readme";
+    changelog = "https://github.com/sjmulder/trickle/blob/${finalAttrs.src.rev}/CHANGELOG.md";
+    longDescription = ''
+      This package provides two programs: trickle and tritty.
+
+      trickle is used as part of a shell pipeline and has very low throughput, hence the name: data trickles through the pipe.
+
+      tritty ("trickle tty") spawns an interactive subterminal with very low throughput. It simulates the experience of using a terminal over a slow connection.
+
+      This is useful for testing the robustness and efficiency of terminal applications. It can uncover problems such as excessive redraws of progress bars, or other inefficient use of the terminal.
+
+      Note that upstream names the package "trickle", but that name was already taken by another package in Nixpkgs.
+    '';
+    # Arguably upstream would pick trickle as the main program, but we can't name it that; see longDescription.
+    # So we pick tritty as the main program instead, for a consistent experience.
```

Comment:
Out of curiosity: If this package were to use `trickle` as the `mainProgram` would this create a collision if this package the the `tricke` package are installed together?

Would `trickle-tty` be "better" name for this package?

Reply:
> If this package were to use `trickle` as the `mainProgram` would this create a collision if this package the the `trickle` package are installed together?

It would create a collision, regardless of `mainProgram`.



> Would `trickle-tty` (or a variation thereof) be a more suitable name for this package?

I prefer not to make up new names, but otherwise I have no strong opinion.

In the context of the previous point though, perhaps it'd be better to split the package between the two commands.
- `tritty`: unique name and no collision with existing `trickle` package
- `trickle-sjmulder`: unambiguous attribute name; possible collision - when installed into a profile with `trickle`

Unfortunately the makefile did not anticipate doing a single command only.

I don't know how realistic the collision is. I'm inclined to keep it simple until someone actually encounters the issue and raises it.

Reply:
> It would create a collision, regardless of `mainProgram`.

Thanks for clarifying, @roberth.

> I prefer not to make up new names, but otherwise I have no strong opinion.

Fair enough 🙂

> Unfortunately the makefile did not anticipate doing a single command only.

Would you find calling the [`trickle` and `tritty` Makefile targets](https://github.com/sjmulder/trickle/blob/master/Makefile#L31-L35) via `makeTarget` and installing the manpage and resulting binary in custom and `installPhase` acceptable?

E.g. for `…/trickle/package.nix`:
```nix
makeTarget = "trickle";

installPhase = ''
  runHook preInstall
  install -Dm755 -t "$out/bin" trickle
  installManPage trickle.1
  runHook postInstall
```

and for `…/tritty/package.nix`:
```nix
makeTarget = "tritty";

installPhase = ''
  runHook preInstall
  install -Dm755 -t "$out/bin" tritty
  installManPage tritty.1
  runHook postInstall
```

> I'm inclined to keep it simple until someone actually encounters the issue and raises it.

That seems reasonable to me.

Reply:
I'd like to have a shared file, because these programs are distributed together and built in the same kind of environment, but unfortunately the by-name rules don't allow that. I guess I could define one as an override of the other.

I believe this is a case where the rules are counterproductive. wdyt @infinisil?

Reply:
Indeed if you want shared files, `pkgs/by-name` doesn't work for that. It's fine to move it away from `pkgs/by-name` for that purpose.